src/qscand.cpp

```diff
@@ -0,0 +1,372 @@
+//			printf("%d: SIGINT. Terminating listener...\n", pid);
+	} else if (childl == 1) {
+		if (debug && !daemonized)
+			printf("%d: SIGINT. Terminating client connection...\n", pid);
+	}
+	term=1;
+#if defined (_WIN32)
+	return 1;
+#endif
+}
+
+#ifdef DAEMON_EN
+int daemonize()
+{
+	pid_t pid;
+	switch (pid = fork()) {
+		case -1:
+			return -1;
+		case 0:
+			openlog("qscand", LOG_PID, LOG_DAEMON);
+			close(0);
+			close(1);
+			close(2);
+			daemonized = 1;
+			break;
+		default:
+			exit(0);
+	}
+	if (setsid() == -1)
+		return -1;
+	return 0;
+}
+#endif
+
+static struct option long_options[] = {
+	{"help",      0, NULL, 'h'},
+#ifdef DAEMON_EN
+	{"nodaemon",  0, NULL, 'n'},
+#endif
+	{"debug",     0, NULL, 'd'},
+	{"iface",     1, NULL, 'i'},
+	{"port",      1, NULL, 'p'},
+#ifdef AUTOPORT_ENABLE
+	{"portauto",  0, NULL, 'a'},
+#endif
+	{0,0,0,0}
+};
+	
+int child_create(int idx)
+{
+	int r;
+	if ((r = thread_create(&childs[idx].tid, NULL, &child_thread, (void*)&childs[idx].arg))) {
+#ifdef DAEMON_EN
+		syslog(LOG_ERR, "Error creating thread!\n");
+#else
+		printf("Error creating thread!\n");
+#endif
+		return r;
+		close(childs[idx].arg.connfd);
+	}
+	return 0;
+}
+
+int main (int argc, char **argv)
+{
+//	pid_t cpid;
+	char *iface=NULL;
+	uint32_t ip;
+	uint16_t port;
+	int listenfd;
+	struct sockaddr_in srvaddr;
+	socklen_t srvaddr_len = sizeof(srvaddr);
+	int      cli_idx;
+#ifdef _WIN32
+	WSADATA WSAdata;
+#endif
+
+	bool portauto=0;
+	bool nodaemon=0;
+	daemonized = 0;
+
+	struct timeval tv;
+	fd_set rd_set;
+	FD_ZERO(&rd_set);
+
+	term=0;
+	clients=0;
+	cli_idx=-1;
+	pid=getpid();
+//	for (int i=0; i<CLIENTS_MAX; i++) cpids[i]=0;
+
+	ip = htonl(INADDR_ANY);
+	port = 0;
+	while (1) {
+		char c = getopt_long(argc, argv, "hndai:p:", long_options, NULL);
+		if (c == -1)
+			break;
+		switch(c) {
+			case 'h':
+				printf("Usage: qscand <options>\n");
+				printf("\n");
+				printf("-h --help           you are reading this:)\n");
+#ifdef DAEMON_EN
+				printf("-n --nodaemon       don't daemonize\n");
+#endif
+				printf("-i --iface <ip>     listen on interface with selected address\n");
+				printf("-p --port <port>    listen selected port\n");
+#ifdef AUTOPORT_ENABLE
+				printf("-a --portauto       autoselect port listen to\n");
+#endif
+				printf("-d --debug          debug\n");
+				printf("\n");	
+				exit(0);
+			case 'n':
+				nodaemon=1;
+#ifndef DAEMON_EN
+				printf("daemon mode not available, option '-n' ignored!\n");
+#endif
+				break;
+			case 'd':
+				debug=1;
+				break;
+			case 'i':
+				iface = optarg; 
+				break;
+			case 'p':
+				{
+					unsigned long lp = strtol(optarg, NULL, 10);
+					if (errno == ERANGE || lp > 65535) {
+						printf("port number out of range!\n");
+					} else {
+						port = lp;
+					}
+				}
+				break;
+#ifdef AUTOPORT_ENABLE
+			case 'a':
+				portauto=1;
+				break;
+#endif
+			default:
+				printf("Invalid option: -%c\n", c);
+				break;
+		}
+	}
+	if (!port && !portauto)
```

Comment:
Codacy found an issue: [Condition '!portauto' is always true](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)